### PR TITLE
NEW Move deployment to GitHub Pages

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Description

Adds a new workflow to build and deploy to GitHub Actions, removes references to Vercel, and makes a small change necessary for the site to work in its new hosting.

## Manual testing steps

You can try this out in your own account if you want - but [here's one I prepared earlier](https://guysartorelli.github.io/github-issue-search-client/) and [here's the deployment](https://github.com/GuySartorelli/github-issue-search-client/actions/runs/9655460058)

## Issues

- https://github.com/silverstripe/github-issue-search-client/issues/139

### CODE REVIEW

File: .github/workflows/deploy-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+on:
+  push:
+    branches: [ "master" ]
+  workflow_dispatch:
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "gh-pages"
+  cancel-in-progress: false
```
Comment (GuySartorelli) [inline .github/workflows/deploy-gh-pages.yml:12]: This isn't strictly necessary, but it was shown in the documentation and seems like a good safeguard to ensure we don't get into a broken state if one action is interrupted and then the subsequent one fails, or something like that.

File: .github/workflows/deploy-gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+permissions: {}
+
+jobs:
+  build:
+    name: Build site
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'silverstripe'
+    permissions:
+      contents: read
```
Comment (GuySartorelli) [inline .github/workflows/deploy-gh-pages.yml:22]: Needed for code checkout

File: .github/workflows/deploy-gh-pages.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          path: "dist/"
+
+  deploy:
+    name: Deploy site
+    needs: build
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'silverstripe'
+    permissions:
+      pages: write
+      id-token: write
```
Comment (GuySartorelli) [inline .github/workflows/deploy-gh-pages.yml:64]: These permissions are documented at https://github.com/actions/deploy-pages?tab=readme-ov-file#security-considerations

File: .nowignore
Comment (GuySartorelli) [inline .nowignore:1]: This file was only for vercel

File: README.md [hunk 4]
```diff
@@ -9,24 +9,18 @@ as well as other repositories that are important for maintenance.
 
 It uses [vue-cli](https://github.com/vuejs/vue-cli/blob/dev/docs/cli.md)
 to simplify local development and production builds for VueJS.
-You can use it on https://github-issue-search-client-silverstripe.vercel.app.
+You can use it on <https://silverstripe.github.io/github-issue-search-client/>.
```
Comment (GuySartorelli) [inline README.md:12]: This URL will change again once we have a solid domain - but it's better than having the vercel URL here in the meantime.

File: src/App.vue
Comment (GuySartorelli) [inline src/App.vue:1]: Just removing the "powered by Vercel" link

File: vue.config.js [hunk 5]
```diff
@@ -8,5 +8,6 @@ module.exports = {
     apollo: {
       lintGQL: false
     }
-  }
+  },
+  publicPath: "./"
```
Comment (GuySartorelli) [inline vue.config.js:12]: Required for the script and style tags to use relative paths, which is necessary for them to load at all in GitHub Pages (at least until we have a solid subdomain - but even then relative will work)

File: .github/workflows/deploy-gh-pages.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Setup GH pages
+        id: pages
+        uses: actions/configure-pages@v5
+
+      - name: Build static files
+        shell: bash
+        run: |
+          cat << EOF > .env
+          VUE_APP_GRAPHQL_ENDPOINT=https://api.github.com/graphql
+          VUE_APP_GRAPHQL_TOKEN=${{ secrets.GRAPHQL_TOKEN_FOR_GH_PAGES }}
```
Comment (GuySartorelli) [inline .github/workflows/deploy-gh-pages.yml:46]: I've moved the same token that Vercel was using into the secrets for this repository.
Note that this token has always been made public by being included in the dist *.js files, so there's no change in attack surface by including it here like this. The .env file isn't included in the artifacts and therefore isn't in the GitHub Pages result - but the token will will be in the dist *.js files as it always has been.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**